infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/BaseLogEntryReader.java

```diff
@@ -0,0 +1,365 @@
+package org.corfudb.infrastructure.logreplication.replication.send.logreader;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.TextFormat;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.common.metrics.micrometer.MeterRegistryProvider;
+import org.corfudb.infrastructure.logreplication.infrastructure.LogReplicationContext;
+import org.corfudb.protocols.logprotocol.OpaqueEntry;
+import org.corfudb.protocols.logprotocol.SMREntry;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.LogReplication;
+import org.corfudb.runtime.exceptions.TrimmedException;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.runtime.view.stream.OpaqueStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.google.protobuf.UnsafeByteOperations.unsafeWrap;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUuidMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLogReplication.generatePayload;
+import static org.corfudb.protocols.service.CorfuProtocolLogReplication.getLrEntryMsg;
+
+@Slf4j
+public abstract class BaseLogEntryReader extends LogEntryReader {
```

Comment:
Not sure if I followed this. From the naming of the classes, isn't LogEntryReader should have extended from baseLogEntry Reader?

Same for BaseSnapshotReader below. Let me know if you think otherwise.